expression/builtin_string.go

```diff
@@ -3019,6 +3019,54 @@ func (b *builtinFormatSig) evalString(row chunk.Row) (string, bool, error) {
 	if isNull || err != nil {
 		return "", isNull, err
 	}
+	if strings.Contains(x, ".") {
+		xArr := strings.Split(x, ".")
+		x1 := xArr[0]
+		x2 := xArr[1]
+		digit, err := strconv.Atoi(d)
+		if err != nil {
+			return "", isNull, err
+		}
+		if len(x2) > digit {
+			t := []byte(x2)
+			carry := false
+			if t[digit] >= '5' {
+				carry = true
+			}
+			if carry == true {
+				for i := digit - 1; i >= 0; i-- {
+					if t[i] == '9' && carry == true {
+						t[i] = '0'
+						carry = true
+					} else if carry == true {
+						t[i] = t[i] + 1
+						carry = false
```

Comment:
Put `carry` check in the for loop, so we can break the loop once `carry` is changed to false?